src/module.c

```diff
@@ -0,0 +1,101 @@
+#include "verify.h"
+#include <node_api.h>
+#include <stdio.h>
+
+napi_value Call_VF_verify(napi_env env, napi_callback_info info) {
+  napi_status status;
+  size_t argc = 6;
+  napi_value argv[argc];
+  status = napi_get_cb_info(env, info, &argc, argv, NULL, NULL);
+
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL, "Failed to parse arguments");
+  }
+
+  // Get the buffer lengths
+  uint64_t pubkey_l;
+  uint64_t document_l;
+  uint64_t signature_l;
+  status = napi_get_value_int64(env, argv[1], (int64_t *)&pubkey_l);
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL, "failed to read pubkey length");
+  }
+  status = napi_get_value_int64(env, argv[3], (int64_t *)&document_l);
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL, "failed to read document length");
+  }
+  status = napi_get_value_int64(env, argv[5], (int64_t *)&signature_l);
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL, "failed to read signature length");
+  }
+
+  void *pubkey;
+  void *document;
+  void *signature;
+
+  status = napi_get_buffer_info(env, argv[0], &pubkey, (size_t *)&pubkey_l);
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL,
+                     "failed to get information about pubkey buffer");
+  }
+  status = napi_get_buffer_info(env, argv[2], &document, (size_t *)&document_l);
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL,
+                     "failed to get information about document buffer");
+  }
+  status =
+      napi_get_buffer_info(env, argv[4], &signature, (size_t *)&signature_l);
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL,
+                     "failed to get information about signature buffer");
+  }
+
+  struct Error *err = NULL;
+  return_t result;
+  result = VF_verify(pubkey, pubkey_l, document, document_l, signature,
+                     signature_l, &err);
+
+  // We should read all of these at some point, and maybe even
+  // give them all the information in the error nodes
```

Comment:
I'm not sure what this means -- is it indicating that only the first error is displayed?

Reply:
I've rephrased this.